stats/src/main/java/io/airlift/stats/cardinality/HyperLogLog.java

```diff
@@ -84,6 +84,9 @@ public void mergeWith(HyperLogLog other)
             ((SparseHll) instance).mergeWith((SparseHll) other.instance);
             instance = makeDenseIfNecessary((SparseHll) instance);
         }
+        else if (instance instanceof DenseHll && other.instance instanceof SparseHll) {
```

Comment:
What about the other way around?

Reply:
In that case, the sparse needs to be converted to dense before the merge can take place. That’s handled by the fall-through case